### DISCUSSION TITLE
tensor in validation_step cannot flow a gradient

### DISCUSSION THREAD

**Discussion by EMD-2026** [Lightning-AI/pytorch-lightning#17906] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I'm trying to calculate the gradient of `pos` from the model output during the validation.
I didn't execute `trainer.fit(lit_model)` Instead, I executed just `trainer.validate(lit_model)`, overriding with

```
    def validation_step(self, batch, batch_idx):
        torch.set_grad_enabled(True)
        pos = batch.pos
        pos.requires_grad_(True)
        print(pos.requires_grad, 0)
        print(pos[0].requires_grad, 1)
        print(pos[:].requires_grad, 2)
        ...
```

I expected the results as
```
True 0
True 1
True 2
```
But I got the results,
```
True 0
False 1
False 2
```
What causes the `False` `requires_grad`?

**Comment by EMD-2026** [comment]
I solved the problem by setting `inference_mode=False`
